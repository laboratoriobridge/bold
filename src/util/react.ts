import { MutableRefObject, Ref } from 'react'

export function setRef<T>(ref: Ref<T>, value: any) {
    if (typeof ref === 'function') {
        ref(value)
    } else if (ref) {
        (ref as MutableRefObject<T>).current = value
    }
}
