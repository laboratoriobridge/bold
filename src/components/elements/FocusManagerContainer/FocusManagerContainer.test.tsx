import * as React from 'react'
import { fireEvent, render, wait } from 'react-testing-library'

import { FocusManagerContainer } from './FocusManagerContainer'

it('should render children', () => {
    const { container } = render(<FocusManagerContainer><span>Children</span></FocusManagerContainer>)
    expect(container).toMatchSnapshot()
})

it('should call onFocusIn prop when focus enter the first element of the container', () => {
    const focusIn = jest.fn()
    const { container } = render(
        <FocusManagerContainer onFocusIn={focusIn}>
            <input id='input1' />
            <input id='input2' />
        </FocusManagerContainer>
    )
    expect(focusIn).not.toHaveBeenCalled()
    fireEvent.focus(container.querySelector('#input1'))
    expect(focusIn).toHaveBeenCalledTimes(1)
    fireEvent.focus(container.querySelector('#input2'))
    expect(focusIn).toHaveBeenCalledTimes(1)
})

it('should call onFocusOut prop when focus leaves the container', async () => {
    const focusOut = jest.fn()
    const { container } = render(
        <FocusManagerContainer onFocusOut={focusOut}>
            <input id='input1' />
            <input id='input2' />
        </FocusManagerContainer>
    )
    expect(focusOut).not.toHaveBeenCalled()

    fireEvent.focus(container.querySelector('#input1'))
    await wait()
    expect(focusOut).not.toHaveBeenCalled()

    fireEvent.focus(container.querySelector('#input2'))
    await wait()
    expect(focusOut).not.toHaveBeenCalled()

    fireEvent.blur(container.querySelector('#input2'))
    await wait()
    expect(focusOut).toHaveBeenCalledTimes(1)
})
