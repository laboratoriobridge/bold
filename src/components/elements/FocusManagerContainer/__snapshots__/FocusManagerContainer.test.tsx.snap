// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render children 1`] = `
<div>
  <div>
    <span>
      Children
    </span>
  </div>
</div>
`;
